rs/consensus/src/consensus/validator.rs

```diff
@@ -956,12 +1035,21 @@ impl Validator {
                 // proposals proceed to be checked normally.
             }
 
+            // Skip block proposals with a disqualified rank. We do this after
+            // checking for a fast-path validation, to avoid getting stuck.
+            if disqualified_ranks
+                .get(proposal.height(), proposal.rank())
+                .is_some()
+            {
+                continue;
```

Comment:
I guess we can't invalidate them because we might need to validate them, in case they get notarized after all?

Reply:
Correct